### PR TITLE
(2707) Exclude non ODA from IATI exports

### PR DESCRIPTION
**Pull request by yndajas**
## Changes in this PR

This filters out non-ODA activities from IATI exports

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by yndajas** [comment]
[Exclude non ODA Activities from IATI exports](https://trello.com/c/zYId3ftX/2707-exclude-non-oda-activities-from-iati-exports)

File: app/services/find_programme_activities.rb [hunk 1]
```diff
@@ -1,16 +1,17 @@
 class FindProgrammeActivities
   include Pundit::Authorization
 
-  attr_accessor :organisation, :user, :fund_code
+  attr_accessor :organisation, :user, :fund_code, :is_oda
 
-  def initialize(organisation:, user:, fund_code: nil)
+  def initialize(organisation:, user:, fund_code: nil, is_oda: [nil, true, false])
```
Comment (Gweaton) [inline app/services/find_programme_activities.rb:6]: Could we make `is_oda` a single boolean here and do this inside `#call`? I found it a little difficult to parse when reading where this is initialized in `programme_activities`

File: app/services/find_project_activities.rb [hunk 2]
```diff
@@ -1,16 +1,17 @@
 class FindProjectActivities
   include Pundit::Authorization
 
-  attr_accessor :organisation, :user, :fund_code
+  attr_accessor :organisation, :user, :fund_code, :is_oda
 
-  def initialize(organisation:, user:, fund_code: nil)
+  def initialize(organisation:, user:, fund_code: nil, is_oda: [nil, true, false])
```
Comment (Gweaton) [inline app/services/find_project_activities.rb:6]: again, see my previous comment about passing an array of values here

File: app/services/find_third_party_project_activities.rb [hunk 3]
```diff
@@ -1,16 +1,17 @@
 class FindThirdPartyProjectActivities
   include Pundit::Authorization
 
-  attr_accessor :organisation, :user, :fund_code
+  attr_accessor :organisation, :user, :fund_code, :is_oda
 
-  def initialize(organisation:, user:, fund_code: nil)
+  def initialize(organisation:, user:, fund_code: nil, is_oda: [nil, true, false])
```
Comment (Gweaton) [inline app/services/find_third_party_project_activities.rb:6]: you know what I'm going to say here!

**Review by Gweaton — COMMENTED** [review]
This looks good - just one thing re: argument type to each of these methods.

File: app/services/find_programme_activities.rb (hunk 1 above)
Comment (Gweaton) [reply app/services/find_programme_activities.rb:6]: Maybe a verbose `include_ispf_non_oda_activities` that defaults to `false` to keep it in sync with the rest of the way the class has been structured?

**Review by Gweaton — APPROVED** [review]
Thanks for the change!

**Pull request merged**